apps/verticalhome/js/app.js

```diff
@@ -150,6 +164,206 @@
 
     },
 
+    saveAllPinnedApps: function(){
+      if(this.deletedApps.length>0){
+
+          for(var x = 0; x < app.pinnedAppsManager.items.length; x++){
+            if(app.pinnedAppsManager.items[x].element.style.display == 'none'){
+              app.pinnedAppsManager.items.splice(x,1);
```

Comment:
It would be easier to understand this code and good for optimization if you keep app.pinnedAppsManager.items in seperate variable